network-api/networkapi/wagtailpages/pagemodels/libraries/detail_page.py

```diff
@@ -0,0 +1,206 @@
+        null=True,
+        blank=False,
+        on_delete=models.SET_NULL,
+        help_text=(
+            "Select a cover image for this article. "
+            "The cover image is displayed on the detail page and all article listings."
+        ),
+    )
+    original_publication_date = models.DateField(
+        null=True,
+        blank=True,
+        help_text="When was the article (not this page) originally published?",
+    )
+    introduction = models.CharField(
+        null=False,
+        blank=True,
+        max_length=300,
+        help_text=(
+            "Provide a short blurb about the article that will be displayed on listing pages and search results."
+        ),
+    )
+    overview = wagtail_fields.RichTextField(
+        null=False,
+        blank=True,
+        features=base_rich_text_options,
+        help_text=(
+            "Provide an overview about the article. "
+            "This can be an excerpt from or the executive summary of the original paper."
+        ),
+    )
+    collaborators = models.TextField(
+        null=False,
+        blank=True,
+        help_text="List all contributors that are not the project leading authors.",
+    )
+
+    content_panels = wagtail_models.Page.content_panels + [
+        image_handlers.FieldPanel("cover_image"),
+        edit_handlers.InlinePanel("links", heading="Article links"),
+        edit_handlers.FieldPanel("original_publication_date"),
+        edit_handlers.FieldPanel("introduction"),
+        edit_handlers.FieldPanel("overview"),
+        edit_handlers.InlinePanel("authors", heading="Authors", min_num=1),
+        edit_handlers.FieldPanel("collaborators"),
+    ]
+
+    translatable_fields = [
+        localize_fields.TranslatableField("title"),
+        localize_fields.SynchronizedField("cover_image"),
+        localize_fields.SynchronizedField("original_publication_date", overridable=False),
+        localize_fields.TranslatableField("links"),
+        localize_fields.TranslatableField("introduction"),
+        localize_fields.TranslatableField("overview"),
+        localize_fields.TranslatableField("authors"),
+        # Promote tab fields
+        localize_fields.SynchronizedField("slug"),
+        localize_fields.TranslatableField("seo_title"),
+        localize_fields.SynchronizedField("show_in_menus"),
+        localize_fields.TranslatableField("search_description"),
+        localize_fields.SynchronizedField("search_image"),
+    ]
+
+    search_fields = wagtail_models.Page.search_fields + [
+        index.SearchField("introduction"),
+        index.SearchField("overview"),
+        index.SearchField("collaborators"),
+        index.FilterField("original_publication_date"),  # For sorting
+        index.RelatedFields(
+            "authors",
+            [
+                index.RelatedFields(
+                    "author_profile",
+                    [index.SearchField("name")],
+                )
+            ],
+        ),
+    ]
+
+    @property
```

Comment:
I've opted to make these properties so that we don't need to pass them onto the context and we can cache them (see the concrete implementations)